### PR TITLE
Include Organelle info in Names and taxonomy section

### PR DESCRIPTION
**Pull request by Swaathik**
## Purpose
https://www.ebi.ac.uk/panda/jira/browse/TRM-27921

## Approach
There are few edge cases where the evidences are not parsed. More info [here](https://www.ebi.ac.uk/panda/jira/browse/TRM-30548). It is taken from references and mapped in our side

## Testing
What test(s) did you write to validate and verify your changes?

## Checklist
- [x] My PR is scoped properly, and “does one thing only”
- [x] I have reviewed my own code
- [x] I have checked that linting checks pass and type safety is respected
- [x] I have checked that tests pass and coverage has at least improved, and if not explained the reasons why

### CODE REVIEW

File: src/uniprotkb/components/entry/NamesAndTaxonomySection.tsx [hunk 1]
```diff
@@ -57,6 +68,58 @@ const NamesAndTaxonomySection = ({ data, primaryAccession }: Props) => {
...
+              if (og.evidences) {
+                evidence = <UniProtKBEvidenceTag evidences={og.evidences} />;
+              } else if (references) {
+                // Mapping the evidences to the corresponding encoding type when the evidences are not parsed in the OG object itself (eg. P09130, P04737)
+                const associatedReferences = references.filter((r) =>
+                  r.referenceComments?.some(
+                    (comment) =>
+                      comment.value.toLowerCase() === og.value.toLowerCase() &&
+                      og.geneEncodingType.toLowerCase() ===
+                        comment.type.toLowerCase()
```
Comment (dlrice) [inline src/uniprotkb/components/entry/NamesAndTaxonomySection.tsx:90]: Minor suggestion to pull out the `og.value.toLowerCase()` and `og.geneEncodingType.toLowerCase()` of the `some` body to avoid repeating `toLowerCase`.

**Review by dlrice — APPROVED** [review]
